Could not find file AssetPackManifestTemplate.plist in "WriteAssetPackManifest" task when run in DevOps

In a .Net Maui project that was migrated from Xamarin, I can run the exact same dotnet publish command on my local Mac and the command completes successfully but when I run the same publish command in Azure DevOps using the DotNetCLI@2 task then I get an error that it cannot find the AssetPackManifestTemplate.plist file during the execution of the [WriteAssetPackManifest](https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.iOS.Tasks/Tasks/WriteAssetPackManifest.cs) task.

### Steps to Reproduce

1. create an .Net Maui app
2. create a Azure DevOps build using the DotNetCoreCLI@2
3. command: "publish"
4. arguments: -c "$(BuildConfiguration)" -f "net7.0-ios" -o $(build.artifactstagingdirectory) -r ios-arm64 --self-contained
5. vmImage: macos-13
6. DotNetVersion: 7.0.403
7. xcode version 15.0.1

### Expected Behavior
UseDotNet@2 task completes successfully

### Actual Behavior
error MSB4018: The "WriteAssetPackManifest" task failed unexpectedly.
error MSB4018: System.IO.FileNotFoundException: Could not find file '/Users/runner/work/1/s/Inform.Forms/obj/Test/net7.0-ios/ios-arm64/ipa/Payload/Inform.Forms.app/AssetPackManifestTemplate.plist'. 

### Environment
Pool: [Azure Pipelines](https://dev.azure.com/b2wsoftware/8be93b52-3f2e-4e86-b29f-4ec214a84418/_settings/agentqueues?poolId=&queueId=40)
Image: macos-13
Agent: Hosted Agent

Starting: Build iOS App

Task         : .NET Core
Description  : Build, test, package, or publish a dotnet application, or run a custom dotnet command
Version      : 2.221.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/dotnet-core-cli

/Users/runner/hostedtoolcache/dotnet/dotnet publish /Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj -c Test -f net7.0-ios -o /Users/runner/work/1/a -r ios-arm64 --self-contained
MSBuild version 17.7.3+4fca21998 for .NET

/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018: The "WriteAssetPackManifest" task failed unexpectedly. [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018: System.IO.FileNotFoundException: Could not find file '/Users/runner/work/1/s/Inform.Forms/obj/Test/net7.0-ios/ios-arm64/ipa/Payload/Inform.Forms.app/AssetPackManifestTemplate.plist'. [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018: File name: '/Users/runner/work/1/s/Inform.Forms/obj/Test/net7.0-ios/ios-arm64/ipa/Payload/Inform.Forms.app/AssetPackManifestTemplate.plist' [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError) [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Func`4 createOpenException) [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access) [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Xamarin.MacDev.PObject.FromFile(String fileName, Boolean& isBinary) in /Users/builder/azdo/_work/1/s/xamarin-macios/external/Xamarin.MacDev/Xamarin.MacDev/PListObject.cs:line 336 [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Xamarin.MacDev.PDictionary.FromFile(String fileName, Boolean& isBinary) in /Users/builder/azdo/_work/1/s/xamarin-macios/external/Xamarin.MacDev/Xamarin.MacDev/PListObject.cs:line 751 [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Xamarin.MacDev.PDictionary.FromFile(String fileName) in /Users/builder/azdo/_work/1/s/xamarin-macios/external/Xamarin.MacDev/Xamarin.MacDev/PListObject.cs:line 739 [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Xamarin.iOS.Tasks.WriteAssetPackManifestTaskBase.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.iOS.Tasks/Tasks/WriteAssetPackManifestTaskBase.cs:line 28 [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Xamarin.iOS.Tasks.WriteAssetPackManifest.Execute() in /Users/builder/azdo/_work/1/s/xamarin-macios/msbuild/Xamarin.iOS.Tasks/Tasks/WriteAssetPackManifest.cs:line 11 [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]
/Users/runner/hostedtoolcache/dotnet/packs/Microsoft.iOS.Sdk/16.4.7129/tools/msbuild/iOS/Xamarin.iOS.Common.targets(534,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/Users/runner/work/1/s/Inform.Forms/Inform.Forms.csproj::TargetFramework=net7.0-ios]

--- a/dotnet/targets/Xamarin.Shared.Sdk.targets
+++ b/dotnet/targets/Xamarin.Shared.Sdk.targets
@@ -233,7 +233,7 @@
 			_CopyResourcesToBundle;
 			_CompileCoreMLModels;
 			_CreatePkgInfo;
-			_CreateAssetPackManifest;
+			_CreateAssetPackManifestMobile;
 			_SmeltMetal;
 			_TemperMetal;
 			_DetectAppManifest;
@@ -273,6 +273,8 @@
 			_PostProcessAppBundle;
 		</CreateAppBundleDependsOn>
 	</PropertyGroup>
+
+	<Target Name="_CreateAssetPackManifestMobile" Condition="'$(_PlatformName)' != 'macOS'" DependsOnTargets="_CreateAssetPackManifest" />
 
 	<!-- PublishTrimmed must be calculated as part of a target because IsMacEnabled on Windows will be set after connecting to the Mac -->
 	<Target Name="_ComputePublishTrimmed">

--- a/dotnet/targets/Xamarin.Shared.Sdk.targets
+++ b/dotnet/targets/Xamarin.Shared.Sdk.targets
@@ -233,6 +233,7 @@
 			_CopyResourcesToBundle;
 			_CompileCoreMLModels;
 			_CreatePkgInfo;
+			_CreateAssetPackManifest;
 			_SmeltMetal;
 			_TemperMetal;
 			_DetectAppManifest;

